packages/components/src/globals/scss/vendor/@carbon/type/scss/_styles.scss

```diff
@@ -468,6 +468,44 @@ $display-04: (
   ),
 ) !default;
 
+/// Used to compare with `$tokens` map so tokens can be reset to their default
+/// @type Map
+/// @access private
+/// @group @carbon/type
+$defaultTokens: (
```

Comment:
@joshblack to my knowledge there's no way to clone a Sass variable. This needs confirmed in a separate PR, but we may need to create two token maps: one that we override, and one used to determine when to reset the overrides.

That simplifies the theming mixin to:

```scss
@mixin carbon--type-theme($themedTokens: $tokens) {
  $tokens: $themedTokens !global;

   @content;

   // Reset to default type tokens after apply in content
  @if $tokens != $defaultTokens {
    @include carbon--type-theme($defaultTokens);
  }
}
```

Reply:
@mattrosno Just wondered - Is cloning a Sass map what you wanted to do? If so, it can be done by map-walk technique.

Reply:
@asudoh I'm not familiar with "map-walk" technique to clone a Sass map. Can you share an example?

Reply:
Actually for simple cloning we don't seem to do map-walking - We can simply do: `map-merge((), $source-map)`